src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -370,7 +396,7 @@ public bool EndsWith(string text, StringComparison comparisonType)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public string Substring(int offset, int length)
         {
-            if (!HasValue || offset < 0 || length < 0 || (uint)(offset + length) > (uint)Length)
+            if (!HasValue || (offset | length) < 0 || (uint)(offset + length) > (uint)Length)
```

Comment:
FYI these changes shouldn't be necessary as this pattern will be addressed by https://github.com/dotnet/runtime/pull/49548, which is pending review.

Reply:
Thanks, good to know (I missed that PR).